### PR TITLE
feat(header-checker-lint): use bot-config-utils

### PR DESCRIPTION
**Pull request by tmatsuo**
part of #1038

With this PR, the bot will start validating the config schema on PRs
and submit failing status checks when the config is invalid.

### CODE REVIEW

**Comment by tmatsuo** [comment]
I also manually tested python-docs-samples config file (it's actually a json file) locally just in case, and it worked.

File: packages/header-checker-lint/test/header-checker-lint.ts [hunk 1]
```diff
@@ -39,15 +43,30 @@ describe('HeaderCheckerLint', () => {
...
     let payload: EventPayloads.WebhookPayloadPullRequest;
+    const sandbox = sinon.createSandbox();
+    let getConfigStub: sinon.SinonStub;
+    let validateConfigStub: sinon.SinonStub;
 
     beforeEach(() => {
       payload = require(resolve(fixturesPath, './pull_request_opened'));
+      getConfigStub = sandbox.stub(botConfigUtilsModule, 'getConfig');
+      // By default, getConfigStub resolves to null.
+      getConfigStub.resolves(null);
```
Comment (chingor13) [inline packages/header-checker-lint/test/header-checker-lint.ts:59]: nit: I might leave this unset, so we don't need to call `reset()` in all the tests that do use config. Then we would explicitly resolve `null` in the test cases that don't have config.

**Review by chingor13 — APPROVED** [review]
Minor (non-blocking) nit

File: packages/header-checker-lint/test/header-checker-lint.ts (hunk 1 above)
Comment (tmatsuo) [reply packages/header-checker-lint/test/header-checker-lint.ts:59]: sounds good

**Review by sofisl — APPROVED** [review]

**Pull request merged**